src/Microsoft.Health.Dicom.Blob/Features/Export/AzureBlobExportSink.cs

```diff
@@ -0,0 +1,166 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Identity;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using EnsureThat;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Blob.Configs;
+using Microsoft.Health.Core;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.Export;
+using Microsoft.Health.Dicom.Core.Models.Common;
+using Microsoft.Health.Dicom.Core.Models.Export;
+
+namespace Microsoft.Health.Dicom.Blob.Features.Export;
+
+internal sealed class AzureBlobExportSink : IExportSink
+{
+    public Uri ErrorHref
+    {
+        get
+        {
+            // Add trailing '/' before concatenating to preserve the container
+            string container = _dest.Uri.GetComponents(UriComponents.SchemeAndServer | UriComponents.Path, UriFormat.Unescaped);
+            if (container[^1] != '/')
+                container += '/';
+
+            return new Uri(new Uri(container, UriKind.Absolute), _output.ErrorFile);
+        }
+    }
+
+    public event EventHandler<CopyFailureEventArgs> CopyFailure;
+
+    private readonly IFileStore _source;
+    private readonly BlobContainerClient _dest;
+    private readonly StreamWriter _errorWriter;
+    private readonly AzureBlobExportFormatOptions _output;
+    private readonly BlobOperationOptions _blobOptions;
+    private readonly JsonSerializerOptions _jsonOptions;
+
+    private const int BlockSize = 2 * 1024 * 1024;
+
+    public AzureBlobExportSink(
+        IFileStore source,
+        BlobContainerClient dest,
+        IOptions<AzureBlobExportFormatOptions> outputOptions,
```

Comment:
If this is not user controlled, should this be passed in or implicit?

Reply:
I have made the output options `internal`, but we can still pass it in as an options as if it's customizable, should we ever need to change this.